lib/setup.coffee

```diff
@@ -128,7 +129,12 @@ module.exports = (app) ->
   switch NODE_ENV
     when 'development'
       reloadAndMount = createReloadable(app, require)
-      app.use reloadAndMount path.join(__dirname, '..', 'apps')
+
+      modules = glob.sync('./react/**/*.js').map (name) => path.resolve(name)
+
+      app.use reloadAndMount(path.join(__dirname, '..', 'apps'), {
+        watchModules: modules
```

Comment:
Oh interesting -- I hadn't thought about this use-case when implementing this, pointing to internal code

Reply:
Actually, I take that back. It doesn't appear to work.

Reply:
In that case you can just `reloadAndMount(path.join(__dirname, 'react')`

Reply:
I'll give this a look today, i think the assumed `node_modules` path under `watchModules` is what's breaking it

Reply:
Appreciate it, sort of works—

An example:

I have a parent-component that gets rendered server-side. I make a change to a sub-component within that parent-component and HMR picks it up and injects it into the browser. But if I refresh, I do not see the changes. Only when I hit save on the parent-component, then I see the changes reflected on the server-side rendered instance.

Reply:
Also in this instance, express-reloadable does log out a notice that the sub-component has changed

Reply:
Can you point me to the exact parent / child? I've noticed this issue in Force when making changes to jade files rendered clientside (seems to be a bug in the webpack loader), but that only applies to that file type, and only sometimes

Reply:
Sure, if you consider:

parent: https://github.com/dzucconi/ervell/blob/b4b880d7c2883d6b2c52a9d2219d33fa2edef23a/react/components/Channel/index.js

child: https://github.com/dzucconi/ervell/blob/b4b880d7c2883d6b2c52a9d2219d33fa2edef23a/react/components/ChannelMetadata/index.js

(or anything under within that)

The parent is called on the server here: https://github.com/dzucconi/ervell/blob/b4b880d7c2883d6b2c52a9d2219d33fa2edef23a/apps/channel/routes.coffee#L23

Reply:
Can you try adding [`recursive: true`](https://github.com/artsy/express-reloadable/blob/master/index.js#L13) to the options?

Reply:
lol: Once I add that if I make a change all of a sudden I get Apollo network errors on every reload. Removing `recursive: true` and everything is fine.

Reply:
Damn... 🤔

Reply:
For now at least just saving the parent file once is much faster than doing a full server restart/recompile!

Reply:
Oh cool, its still possible to run Ervell locally, was able to get it booted -- gonna look at those two spots

Reply:
Ok one last little thing and if that doesn't work not too sure for now (can also glob it instead of the extra `filter`, but you get the idea):

Re-add `recursive: true` and then 

```
    modules = glob.sync('./react/**/*.js')
      .filter (name) => !name.includes('react/apollo') # or name.includes('react/components')
      .map (name) => path.resolve(name)
```

Reply:
No dice. Still getting the same "Network error: only absolute urls are supported"

Reply:
Ah bummer, I'll keep an eye out for this in Force and if make sure to open a PR over here if I find a fix. Unfortunately `express-reloadable` is kinda a leap of faith since its messing with module caches directly without knowledge of what's going on in other libs